src/zeek-setup.cc

```diff
@@ -778,9 +779,15 @@ SetupResult setup(int argc, char** argv, Options* zopts)
 		if ( options.event_trace_file )
 			etm = make_unique<EventTraceMgr>(*options.event_trace_file);
 
+		// Parsing involves reading input files, including any input
+		// interactively provided by the user at the console. Temporarily
+		// undo the signal mask to allow ctrl-c. Ideally we'd do this only
+		// when we actually end up reading interactively from stdin.
```

Comment:
Could we check `isatty(0)`?

Reply:
Ah — yes, it's certainly possible to check we're reading from stdin when that happens. I was more concerned about whether we want to start adjusting the signal mask in more source files because the startup is already pretty complex. The determination that we're reading from stdin lives over in `load_files()` in scan.l.